### PR TITLE
run the 6+ upgrade scripts as the PF user

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
run the 6+ upgrade scripts as the PF user

# Impacts
Upgrade scripts

# Issue
part of the fix for #2378 

# Delete branch after merge
YES

# NEWS file entries
## Bug Fixes
* Run the 6.0+ upgrade scripts as the pf user to prevent permissions issues after running them

### CODE REVIEW

**Comment by fdurand** [comment]
Agree to merge

File: lib/pf/util.pm [hunk 1]
```diff
@@ -1352,6 +1353,17 @@ sub ping {
...
 }
 
+=head2 run_as_pf
+
+Sets the UID and GID of the currently running process to pf
+
+=cut
+
+sub run_as_pf {
+    my (undef, undef,$uid,$gid) = getpwnam('pf');
```
Comment (jrouzierinverse) [inline lib/pf/util.pm:1363]: Check if current user if already pf before running setgid and setuid.

File: lib/pf/util.pm [hunk 2]
```diff
@@ -1352,6 +1353,17 @@ sub ping {
...
+    setgid($gid);
```
Comment (jrouzierinverse) [inline lib/pf/util.pm]: Check for errors and log them

File: lib/pf/util.pm [hunk 3]
```diff
@@ -1352,6 +1353,17 @@ sub ping {
...
+    setuid($uid);
```
Comment (jrouzierinverse) [inline lib/pf/util.pm]: Check for errors and log them

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by julsemaan** [comment]
Should be all polished up, ready for another review

File: lib/pf/util.pm [hunk 4]
```diff
@@ -1352,6 +1353,43 @@ sub ping {
...
+
+Sets the UID and GID of the currently running process to pf
+
+=cut
+
+sub run_as_pf {
+    # Early return if we're already running as pf
+    return $TRUE if(running_as eq "pf");
+
+    my (undef, undef,$uid,$gid) = getpwnam('pf');
```
Comment (jrouzierinverse) [inline lib/pf/util.pm]: Just compare $< to $uid instead of using running_as since the environmental variables could be modified.

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by julsemaan** [comment]
done

**Review by jrouzierinverse — APPROVED** [review]

**Pull request merged**